### PR TITLE
[LM-9366] Scroll Listener

### PR DESCRIPTION
**Pull request by arnavr15**
## Description

**Features**
- Added dual side scroll listener

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: ChatSX/screens/ChatRoom/index.tsx [hunk 1]
```diff
@@ -851,39 +860,138 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
...
-  async function paginatedData(newPage: number) {
+  // Function calls paginatedConversationsEnd action which internally calls getConversations to update conversation array with the new data.
+  async function endOfPaginatedData() {
     let payload = {
       chatroomID: chatroomID,
       conversationID: conversations[conversations.length - 1]?.id,
-      scrollDirection: 0,
-      paginateBy: 50,
+      scrollDirection: 0, //scroll up -> 0
+      paginateBy: 10,
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:869]: reset it to 50

File: ChatSX/screens/ChatRoom/index.tsx [hunk 2]
```diff
@@ -851,39 +860,138 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
...
+    return response;
+  }
+
+  // Function calls paginatedConversationsStart action which internally calls getConversations to update conversation array with the new data.
+  async function startOfPaginatedData() {
+    let payload = {
+      chatroomID: chatroomID,
+      conversationID: conversations[0]?.id,
+      scrollDirection: 1, //scroll down -> 1
+      paginateBy: 10,
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:884]: reset it to 50

File: ChatSX/screens/ChatRoom/index.tsx [hunk 3]
```diff
@@ -851,39 +860,138 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
...
+    if (res.conversations.length == 0) {
+      setShouldLoadMoreChatEnd(false);
+    }
+
+    if (!!res) {
+      setIsLoading(false);
+    }
+  };
+
+  const scrollToVisibleIndex = (ind: number) => {
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:908]: rename to `index`

File: ChatSX/screens/ChatRoom/index.tsx [hunk 4]
```diff
@@ -851,39 +860,138 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
...
+
   // function shows loader in between calling the API and getting the response
-  const loadData = async (newPage: number) => {
+  const startLoadData = async () => {
     setIsLoading(true);
-    const res = await paginatedData(newPage);
+    const res = await startOfPaginatedData();
+
+    // To check if its the start of list (bottom of list in our case)
     if (res.conversations.length == 0) {
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:923]: add option chaining

File: ChatSX/screens/ChatRoom/index.tsx [hunk 5]
```diff
@@ -851,39 +860,138 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
...
-      setShouldLoadMoreChat(false);
+      setShouldLoadMoreChatStart(false);
     }
+
     if (!!res) {
+      const len = res.conversations.length;
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:928]: add option chaining

File: ChatSX/screens/ChatRoom/index.tsx [hunk 6]
```diff
@@ -1805,6 +1918,7 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
   return (
     <View style={styles.container}>
       <FlashList
+        estimatedItemSize={30}
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:1921]: Why 30?

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: ChatSX/screens/ChatRoom/index.tsx (hunk 6 above)
Comment (arnavr15) [reply ChatSX/screens/ChatRoom/index.tsx:1921]: will make it 50

File: ChatSX/screens/ChatRoom/index.tsx [hunk 7]
```diff
@@ -838,6 +838,25 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
     }
   }, [isFocused]);
 
+  useEffect(() => {
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:841]: add comments what is use of this `useEffect()`

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

**Review by ishaan1607 — APPROVED** [review]

**Pull request merged**